Automattic/vip-go-mu-plugins
# Add scheduled user-role backup and management CLI

## Description

Adds 1) a cron job to regularly backup user roles to a secondary option and 2) a CLI to manage listing/viewing/restoring those backups to the primary.

Why? In some cases of option corruption, user roles are lost and restoration is complicated and/or slow. This intends to make it quick and easy.

Note: this deals with the `wp_user_roles` option, not the assignments to users.

## Changelog Description

### Add User Role Backups

[tbd]

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Begin by backing up current roles, then alter roles and backup again:
```
wp cron event run 'Automattic\VIP-backup_roles'
wp role delete editor
wp role delete author
wp role delete contributor
wp cron event run 'Automattic\VIP-backup_roles'
```

A couple options will now be listed in `wp vip role-backup list`

1. optionally view a backup: `wp vip role-backup view [<time_key>]`
2. and restore: `wp vip role-backup restore [<time_key>]`
3. and confirm: `wp role list`
